admiral/pkg/clusters/virtualservice_routing.go

```diff
@@ -1555,8 +1569,53 @@ func addUpdateRoutingDestinationRule(
 		ctxLogger.Infof(common.CtxLogFormat, "addUpdateRoutingDestinationRule",
 			drName, util.IstioSystemNamespace, sourceCluster, "destinationrule created successfully")
 
+		rc.DestinationRuleController.Cache.Put(newDR)
+
+	}
+
+	return nil
+}
+
+// processSlowStartConfig handles the processing of slow start traffic configuration
+// and applies warmup duration to destination rule if found
+func processSlowStartConfig(remoteRegistry *RemoteRegistry, ctxLogger *log.Entry, assetKey string, workloadEnvKey string,
+	drObj *networkingV1Alpha3.DestinationRule, drName string, sourceCluster string) error {
+	assetConfigMap := remoteRegistry.AdmiralCache.SlowStartConfigCache.Get(assetKey)
+	if assetConfigMap == nil {
+		return nil
 	}
 
+	// Use a function to allow breaking from the Range method
+	var warmupDurationValue int64
+	var err error
+	assetConfigMap.Range(func(envKey string, envConfig *common.Map) {
```

Comment:
hey man, sorry I'm probably not understanding this. Looks like we are looping over assetConfig which has a map of workloadEnv and each workloadEnv has duration associated with it. But within the loop we keep overriding `warmupDurationValue` for each workloadEnv. 
Do all workloadEnv have the same warmupDuration? If yes, then we can just get one and break the loop right?

Reply:
Good catch, added a variable to exit early.